File: Streamdeck +/initialise details.py
from StreamDeck.DeviceManager import DeviceManager

def print_deck_info(index, deck):
    key_image_format = deck.key_image_format()
    touchscreen_image_format = deck.touchscreen_image_format()
    
    flip_description = {
        (False, False): "Not mirrored",
        (True, False): "Mirrored horizontally",
        (False, True): "Mirrored vertically",
        (True, True): "Mirrored horizontally and vertically"
    }
    
    print("Deck {} - {}.".format(index, deck.deck_type()))
    print("\t - ID: {}".format(deck.id()))
    print("\t - Serial: '{}'".format(deck.get_serial_number()))
    print("\t - Firmware version: '{}'".format(deck.get_firmware_version()))
    print("\t - Key Count: {} (in a {} x {} grid)".format(
        deck.key_count(), 
        deck.key_layout()[0], 
        deck.key_layout()[1]))
    
    if deck.is_visual():
        print("\t - Key Images: {} x {} pixels, {} format, rotated {} degrees, {}".format(
            key_image_format['size'][0], 
            key_image_format['size'][1], 
            key_image_format['format'], 
            key_image_format['format'], 
            key_image_format['rotation'], 
            flip_description[key_image_format['flip']]))
        
        if deck.is_touch():
            print("\t - Touchscreen: {} x {} pixels, {} format, rotated {} degrees, {}".format(
                touchscreen_image_format['size'][0], 
                touchscreen_image_format['size'][1], 
                touchscreen_image_format['format'], 
                touchscreen_image_format['rotation'], 
                flip_description[touchscreen_image_format['flip']]))
    else:
        print("\t - No images supported.")
        
if __name__ == "__main__":
    streamdecks = DeviceManager().enumerate()
    
    print("Found {} Stream Decks.\n".format(len(streamdecks)))
    
    for index, deck in enumerate(streamdecks):
        deck.open()
        deck.reset()
        
        print_deck_info(index, deck)
        
        deck.close()
        

File: Streamdeck +/initialisedetails.js
const HID = require('node-hid');

function printDeckInfo(index, deck) {
    const keyImageFormat = deck.keyImageFormat();
    const touchscreenImageFormat = deck.touchscreenImageFormat();

    const flipDescription = {
        'false,false': "Not mirrored",
        'true,false': "Mirrored horizontally",
        'false,true': "Mirrored vertically",
        'true,true': "Mirrored horizontally and vertically"
    };

    console.log(`Deck ${index} - ${deck.deckType()}.`);
    console.log(`\t - ID: ${deck.id()}`);
    console.log(`\t - Serial: '${deck.getSerialNumber()}'`);
    console.log(`\t - Firmware version: '${deck.getFirmwareVersion()}'`);
    console.log(`\t - Key Count: ${deck.keyCount()} (in a ${deck.keyLayout()[0]} x ${deck.keyLayout()[1]} grid)`);

    if (deck.isVisual()) {
        console.log(`\t - Key Images: ${keyImageFormat.width} x ${keyImageFormat.height} pixels, ${keyImageFormat.format} format, rotated ${keyImageFormat.rotation} degrees, ${flipDescription[`${keyImageFormat.flipX},${keyImageFormat.flipY}`]}`);
        console.log(`\t - Touchscreen Images: ${touchscreenImageFormat.width} x ${touchscreenImageFormat.height} pixels, ${touchscreenImageFormat.format} format`);
    }
}

function main() {
    const devices = HID.devices();
    const streamDecks = devices.filter(device => device.vendorId === 0x0fd9 && device.productId === 0x0084); // Replace with actual vendorId and productId

    streamDecks.forEach((device, index) => {
        const deck = new HID.HID(device.path);
        printDeckInfo(index, deck);
        deck.close();
    });
}

main();